baremetal/metal3data_manager_test.go

```diff
@@ -1184,11 +1184,12 @@ var _ = Describe("Metal3Data manager", func() {
 			},
 			expectRequeue: true,
 		}),
-		Entry("IPClaim with deletion timestamp", testCaseGetAddressFromPool{
+		Entry("Old IPClaim with deletion timestamp", testCaseGetAddressFromPool{
 			m3d: &infrav1.Metal3Data{
 				ObjectMeta: testObjectMeta(metal3DataName, namespaceName, ""),
 			},
 			poolName:        testPoolName,
+			poolRef:         corev1.TypedLocalObjectReference{Name: testPoolName},
```

Comment:
This test was not working properly when the `poolRef` was missing. It just returned early because it could not find the IPClaim and asked for requeue. Now it is working as expected.